File: src/main/java/com/procurement/access/dao/TenderProcessDaoImpl.java
package com.procurement.access.dao;

import com.datastax.driver.core.Row;
import com.datastax.driver.core.Session;
import com.datastax.driver.core.Statement;
import com.datastax.driver.core.querybuilder.Insert;
import com.procurement.access.model.entity.TenderProcessEntity;
import org.springframework.stereotype.Service;

import static com.datastax.driver.core.querybuilder.QueryBuilder.*;

@Service
public class TenderProcessDaoImpl implements TenderProcessDao {

    private static final String TENDER_TABLE = "access_tender";
    private static final String CP_ID = "cp_id";
    private static final String TOKEN = "token_entity";
    private static final String STAGE = "stage";
    private static final String CREATED_DATE = "created_date";
    private static final String OWNER = "owner";
    private static final String JSON_DATA = "json_data";

    private final Session session;

    public TenderProcessDaoImpl(final Session session) {
        this.session = session;
    }

    @Override
    public void save(final TenderProcessEntity entity) {
        final Insert insert = insertInto(TENDER_TABLE);
        insert.value(CP_ID, entity.getCpId())
                .value(TOKEN, entity.getToken())
                .value(OWNER, entity.getOwner())
                .value(STAGE, entity.getStage())
                .value(CREATED_DATE, entity.getCreatedDate())
                .value(JSON_DATA, entity.getJsonData());
        session.execute(insert);
    }

    @Override
    public TenderProcessEntity getByCpIdAndStage(final String cpId, final String stage) {
        final Statement query = select()
                .all()
                .from(TENDER_TABLE)
                .where(eq(CP_ID, cpId))
                .and(eq(STAGE, stage)).limit(1);
        final Row row = session.execute(query).one();
        if (row != null)
            return new TenderProcessEntity(
                    row.getString(CP_ID),
                    row.getUUID(TOKEN),
                    row.getString(OWNER),
                    row.getString(STAGE),
                    row.getTimestamp(CREATED_DATE),
                    row.getString(JSON_DATA));
        return null;

    }
}

File: src/main/kotlin/com/procurement/access/dao/TenderProcessDaoImpl.kt
package com.procurement.access.dao

import com.datastax.driver.core.Session
import com.datastax.driver.core.querybuilder.QueryBuilder.*
import com.procurement.access.model.entity.TenderProcessEntity
import org.springframework.stereotype.Service

interface TenderProcessDao {

    fun save(entity: TenderProcessEntity)

    fun getByCpIdAndStage(cpId: String, stage: String): TenderProcessEntity?

}

@Service
class TenderProcessDaoImpl(private val session: Session) : TenderProcessDao {

    override fun save(entity: TenderProcessEntity) {
        val insert = insertInto(TENDER_TABLE)
        insert.value(CP_ID, entity.cpId)
                .value(TOKEN, entity.token)
                .value(OWNER, entity.owner)
                .value(STAGE, entity.stage)
                .value(CREATED_DATE, entity.createdDate)
                .value(JSON_DATA, entity.jsonData)
        session.execute(insert)
    }

    override fun getByCpIdAndStage(cpId: String, stage: String): TenderProcessEntity? {
        val query = select()
                .all()
                .from(TENDER_TABLE)
                .where(eq(CP_ID, cpId))
                .and(eq(STAGE, stage)).limit(1)
        val row = session.execute(query).one()
        return if (row != null) TenderProcessEntity(
                row.getString(CP_ID),
                row.getUUID(TOKEN),
                row.getString(OWNER),
                row.getString(STAGE),
                row.getTimestamp(CREATED_DATE),
                row.getString(JSON_DATA)) else null

    }

    companion object {
        private val TENDER_TABLE = "access_tender"
        private val CP_ID = "cp_id"
        private val TOKEN = "token_entity"
        private val STAGE = "stage"
        private val CREATED_DATE = "created_date"
        private val OWNER = "owner"
        private val JSON_DATA = "json_data"
    }
}
